src/Install.php

```diff
@@ -396,12 +396,23 @@ private static function update() {
 						array(
 							'per_page' => 1,
 							'hook'     => 'woocommerce_run_update_callback',
-							'search'   => json_encode( array( $update_callback ) ),
+							'search'   => wp_json_encode( array( $update_callback ) ),
 							'group'    => 'woocommerce-db-updates',
+							'status'   => 'pending',
 						)
 					);
 
-					if ( empty( $pending_jobs ) ) {
+					$complete_jobs = WC()->queue()->search(
+						array(
+							'per_page' => 1,
+							'hook'     => 'woocommerce_run_update_callback',
+							'search'   => wp_json_encode( array( $update_callback ) ),
+							'group'    => 'woocommerce-db-updates',
+							'status'   => 'complete',
+						)
+					);
+
+					if ( empty( $pending_jobs ) && empty( $complete_jobs ) ) {
```

Comment:
Why was this change needed? Are we actually cleaning up completed jobs?

Reply:
Same question.  I *think* this works because if a job has recently completed with the same name, then we don't need to run it again.

If this is the case, I think we could remove the `status` property entirely and just search the queue once.  The API makes it seem like there are only the options of `complete` and `pending` so we could just not limit by status.

Reply:
>The API makes it seem like there are only the options of complete and pending so we could just not limit by status.

To add on this, there are other statuses used, so I'm not 100% sure if the `search()` method only returns `complete` and `pending` as the docs seem to suggest or if it will include all.

Reply:
> the docs seem to suggest or if it will include all.

Yes, including `failed` and `canceled`. We want to schedule the update if the existing action has either of those statuses.